### PR TITLE
Add a bash script to generate tutorial examples.

### PR DESCRIPTION
**Pull request by hferee**
And adapt wordfreq and simple_bst to it.
Issue #350

### CODE REVIEW

**Comment by xrchz** [comment]
Also, can we make the `Holmakefile` in `tutorial/solutions` have a target that generates the exercises and copies them to the right place. So running `Holmake` in `tutorial/solutions` should produce the updates automatically.

**Comment by hferee** [comment]
* I don't see any trailing whitespaces in these files.
* The produced files are slightly different because the script can't uncomment code, only remove sets of lines. (see the above diff)
* `Holmake` now produces the files in the parent directory, as it is currently.

**Comment by xrchz** [comment]
sorry: I think I got confused by some extra blank lines

now it looks like maybe just the `make_ex.sml` file is missing?

**Comment by hferee** [comment]
Indeed. Here it is.

File: tutorial/solutions/make_ex.sml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  let val src = TextIO.openIn(fname)
+      (* In CakeML, BadFilename should take the filename as string *)
+      val dst = TextIO.openOut("../" ^ fname)
+      (* CakeMl's openOut is probably not correct on unexisting files *)
+      val _ = copy_until_solution_start src dst
+      val _ = TextIO.closeIn(src)
+      val _ = TextIO.closeOut(dst)
+  in () end;
+
+(* exists in CakeML? *)
```
Comment (xrchz) [inline tutorial/solutions/make_ex.sml]: yes, it's `List.app`

File: tutorial/solutions/make_ex.sml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  case l of
+       [] => ()
+     | h :: t => (f h; iter f t)
+
+fun main () = let
+  val args = CommandLine.arguments ()
+  val _ = iter handle_file args
+  in () end;
+
+(* for a CakeML version:
```
Comment (xrchz) [inline tutorial/solutions/make_ex.sml]: this rename will happen in #354

**Pull request merged**